code/server/src/ServerModule.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+ * author: Elisa Maiettini
+ * date: 26/11/2016
+ */
+#include<ServerModule.hpp>
+
+using namespace std;
+using namespace yarp::os;
+
+bool ServerModule::respond(const Bottle &command, Bottle &reply){
+	int tested_N;
+	cout << "ServerModule: waiting message from client.."<< endl;
+
+	if(command.get(0).asInt()==COLLATZ_VOCAB_REQ_ITEM){
+		tested_N = command.get(1).asInt();
+		cout << "ServerModule: received message of tested number:" << tested_N << endl;
+
+		//updating CNT and FIFO
+		cout << "ServerModule: updating CNT and FIFO.." << endl;
+		CNT++;
+		fifo_handler->pushBack(CNT);
+
+		//choosing an other pair (N,T)
+		cout << "ServerModule: choosing an other pair (N,T).." <<endl;
+		int new_N = CNT;
+		cout << "CNT: " << CNT << " ";
+
+		int new_T = ((fifo_handler->getHead()) - 1);
+		cout << "HEAD - 1 : " << new_T <<endl;
+
+		//eresing tested number
+		fifo_handler->eraseTested(tested_N);
+
+		//preparing new bottle for the client
+		cout << "ServerModule: preparing bottle for the client.."<< endl;
+		reply.addInt(COLLATZ_VOCAB_ITEM);
+		reply.addInt(new_N);
+		reply.addInt(new_T);
+
+	}
+	return true;
+}
+
+bool ServerModule::updateModule(){
+	return true;
+}
+
+bool ServerModule::configure(yarp::os::ResourceFinder &rf){
+	CNT = 1;
+	int ms_period = 1000;
+
+  string port_name = rf.find("name").asString().c_str();
```

Comment:
usually when using the RF to assign properties you can provide also a default value to be used when such property is not provided, e.g.:

    ms_period = rf.check("rate", Value(1000)).asInt();